### ISSUE TITLE
non-ASCII characters in patches not allowed.

### ISSUE THREAD

**Issue by isuruf** [conda/conda-build#1797]
I get an error when there are non-ASCII characters in a patch.
```
  File "/home/isuru/miniconda3_2/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/cli/main_build.py", line 322, in main
    execute(sys.argv[1:])
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/cli/main_build.py", line 313, in execute
    noverify=args.no_verify)
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/api.py", line 97, in build
    need_source_download=need_source_download, config=config)
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/build.py", line 1484, in build_tree
    config=config)
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/build.py", line 983, in build
    config=config)
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/render.py", line 86, in parse_or_try_download
    source.provide(metadata, config=config)
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/source.py", line 529, in provide
    apply_patch(src_dir, join(metadata.path, patch), config, git)
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/source.py", line 460, in apply_patch
    files, is_git_format = _get_patch_file_details(path)
  File "/home/isuru/miniconda3_2/lib/python3.5/site-packages/conda_build/source.py", line 444, in _get_patch_file_details
    for l in f.readlines():
  File "/home/isuru/miniconda3_2/lib/python3.5/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa3 in position 8127: invalid start byte
```

**Comment by isuruf** [comment]
Changing https://github.com/conda/conda-build/blob/dcc74eb815209cdcf98e0b9681f3b1456983ed93/conda_build/source.py#L444
to `with io.open(path, errors="ignore") as f:` works

**Comment by msarahan** [comment]
fixed by #1802 - closing

**Comment by isuruf** [comment]
Thanks
